rules/terraformrules/terraform_empty_list_equality_test.go

```diff
@@ -41,6 +41,34 @@ resource "aws_db_instance" "mysql" {
 				},
 			},
 		},
+		{
+			Name: "comparing with [] is not recommended (mixed with other conditions)",
+			Content: `
+resource "aws_db_instance" "mysql" {
+	count = true == true || false != true && (false == false || [] == []) ? 1 : 0
+	instance_class = "m4.2xlarge"
+}`,
+			Expected: tflint.Issues{
+				{
```

Comment:
This doesn't seem like a correct result. The expression that is invalid is the comparison, with both left, right, and a comparator. In the case of `[] == []` that is only one invalid expression and should be one issue. Emitting duplicate issues should be a hint that something's off.

Reply:
Just used a map to store the ranges and to avoid emitting multiple issues with the same range. Now it will just report once in case we have `[] == []`